tests/test_co2eq_parameters.py

```diff
@@ -254,8 +254,12 @@ class CO2eqParametersDirectAndLifecycleMixin:
 
     @classmethod
     def check_emission_factors(cls, callback):
-        """Apply the callback to each emission factor in the 'emissionFactors'
-        object.
+        """Apply the callback to each item in the 'emissionFactors' object.
+
+        The callback is called with the mode and factors for both the defaults
+        and the zone-specific overrides; each `factors` object is either a list
+        of factor dicts or a single factor dict, and each factor dict has the
+        key "value", and possibly other keys such as "source".
         """
         emission_factors = cls.parameters["emissionFactors"]
```

Comment:
Pylance is reporting that parameters is unknown on the type `Type[CO2eqParametersDirectAndLifecycleMixin]` so we have to take care of that if we want to add any form of type checking in the future.

Reply:
Done.

Reply:
Note that many of the warnings were missing assert methods since the mixin class was not a subclass of TestCase.

One of the simplest ways to deal with this was to make it a subclass of TestCase; after doing that however it is found automatically by the test runner so to prevent unexpected behavior, in the latest revision I nested it inside of another class, like in https://stackoverflow.com/questions/1323455/python-unit-test-with-base-and-sub-class